test/test.cpp

```diff
@@ -17,13 +17,28 @@ TEST(basic, basic_test) {
   // Assert
   ASSERT_EQ(5, c);
 }
+TEST(bfs, check_result_vec) {
+  const std::vector<int> vec = {1, 2, 3, 4};
+  Graph graph(5);
+  LayerExample a1(0, kInput);
+  LayerExample a2(1, kFullyConnected);
+  LayerExample a3(2, kFullyConnected);
+  LayerExample a4(3, kDropout);
+  graph.setInput(a1, vec);
+  graph.makeConnection(a1, a2);
+  graph.makeConnection(a2, a4);
+  graph.makeConnection(a1, a3);
+  graph.inference();
+  std::vector<int> vec1 = graph.getOutput();
```

Comment:
looks good, but small comments
1. please use setOutput (for multiples outputs or difficult output)
2. remove initialise indexes with users (please use autogenerated indexes into graph) for minimise user's mistakes

example:
```
const std::vector<int> vec_in = {1, 2, 3, 4};
std::vector<int> vec_out(4);
Graph graph(5);
LayerExample a1(kInput);
LayerExample a2(kFullyConnected);
LayerExample a3(kFullyConnected);
LayerExample a4(kDropout);
LayerExample a5(kOutput);
graph.setInput(a1, vec_in);
graph.setOutput(a5, vec_output.data()); 
graph.makeConnection(a1, a2);
graph.makeConnection(a2, a4);
graph.makeConnection(a1, a3);
graph.inference();
```